src/prif.f90

```diff
@@ -1031,4 +1027,12 @@ module subroutine prif_atomic_ref_logical_indirect(image_num, atom_remote_ptr, v
     integer(c_intmax_t) :: lcobounds(15), ucobounds(15)
   end type
 
+  type, private :: team_data
+    type(c_ptr) :: gex_team
+    type(c_ptr) :: heap_mspace
+    integer(c_intptr_t) :: heap_start
+    type(prif_team_type), pointer :: parent_team
```

Comment:
I suspect we actually want the `parent_team` component to point directly to the corresponding `team_data` object, otherwise we'll be forced to allocate and leak a second object. Given the `team_data` pointer, a corresponding `prif_team_type` for the parent team can always be manufactured on-demand.

```suggestion
    type(team_data), pointer :: parent_team
```